src/index.js

```diff
@@ -7,133 +7,119 @@
-        // which is represented by the sole `id` property.
-        // https://www.w3.org/TR/did-core/#did-document-properties
-        if (didResolutionResult.didDocumentMetadata.deactivated) {
+        } else if (didDocumentMetadata.deactivated) {
+          console.info(`\n❌ DID ${did} has been disabled`)
           // sending a 410 according to https://w3c-ccg.github.io/did-resolution/#bindings-https
           res.status(410)
-          didResolutionResult.didDocument = {
-            id: did
-          }
-          if (isJsonLd) {
-            didResolutionResult.didDocument['@context'] = [DID_DOC_CONTEXT]
-          }
-        } else if (resolvedDidDetails && resolvedDidDetails.details) {
-          didResolutionResult.didDocument = Did.exportToDidDocument(
-            resolvedDidDetails.details,
-            isJsonLd ? 'application/ld+json' : 'application/json'
-          )
-          // TODO: This will be added by the SDK automatically once support for the new context is added (most likely 0.30.x).
-          didResolutionResult.didDocument['@context'].push(KILT_DID_CONTEXT)
-
-          if (
-            hasWeb3Names() &&
-            resolvedDidDetails.details instanceof Did.FullDidDetails
-          ) {
-            // check for web3name
-            console.info(`\n🔍 Performing Web3Name lookup for ${did}`)
-            const w3n = await Did.Web3Names.queryWeb3NameForDid(did)
-            if (w3n) {
-              console.info(`   🦸 DID is associated with Web3Name "${w3n}"`)
-              didResolutionResult.didDocument.alsoKnownAs = [`w3n:${w3n}`]
-            } else {
-              console.info(
-                `   ❌ DID is not currently associated with a Web3Name`
-              )
-            }
-          }
         }
 
-        const response =
-          responseContentType === DID_RESOLUTION_RESPONSE_MIME
-            ? didResolutionResult
-            : didResolutionResult.didDocument
-
-        console.info('\n← Responding with:')
-        console.info(JSON.stringify(response, null, 2))
+        // 3. build response according to requested MIME
+        // add json-ld contexts to DID document if json-ld is requested
+        if (didDocument && isJsonLd) {
+          didDocument['@context'] = [W3C_DID_CONTEXT_URL, KILT_DID_CONTEXT_URL]
+        }
 
-        res.contentType(responseContentType).send(response)
+        res.status(200)
       } catch (error) {
         console.error(
           '\n🚨 Could not satisfy request because of the following error:'
         )
         console.error(`${error}`)
-        res.sendStatus(500)
+        res.status(500)
+        didDocument = null
+        didDocumentMetadata = {}
+        didResolutionMetadata = {
+          error: 'internalError',
+          errorMessage: String(error)
+        }
       } finally {
+        // create response body depending on MIME type
+        let response
+        if (responseContentType === DID_RESOLUTION_RESPONSE_MIME) {
```

Comment:
What is this mime type?

Reply:
the mime type for the structure of did document and metadata called 'did resolution result'. It's defined in the did resolution specs